sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py

```diff
@@ -314,14 +288,16 @@ def transform_outbound_single_message(message, message_type):
     :rtype: EventData
     """
     try:
-        # EventData
         # pylint: disable=protected-access
-        return message._to_outgoing_message()  # type: ignore
+        # EventData
+        message._message = to_outgoing_amqp_message(message.raw_amqp_message)
+        return message  # type: ignore
     except AttributeError:
-        # AmqpAnnotatedMessage
         # pylint: disable=protected-access
+        # AmqpAnnotatedMessage
+        amqp_message = to_outgoing_amqp_message(message)
         return message_type._from_message(
-            message=message._to_outgoing_amqp_message(), raw_amqp_message=message  # type: ignore
+            message=amqp_message, raw_amqp_message=message  # type: ignore
```

Comment:
yeah sorry, naming everything message is def confusing and probably was not the best choice 😅. I can add more comments here to describe what's happening.
1) In the try block above (case: message=EventData), we want to create the outgoing uamqp/pyamqp internal Message object. All the changes that are made by the user to EventData properties are persisted on the EventData.raw_amqp_message object (AmqpAnnotatedMessage), and _not_ on the internal EventData._message object (uamqp/pyamqp Message). The `AmqpTransport.to_outgoing_amqp_message` takes the AmqpAnnotatedMessage, and returns the corresponding uamqp/pyamqp Message. The internal EventData._message is set to this outgoing uamqp/pyamqp Message object. The EventData._message objects will be sent.
2) In the except block here (case: message=AmqpAnnotatedMessage, message_type=EventData), we call EventData._from_message, which will create the EventData object from two args:
  1. `message: uamqp/pyamqp Message` param. In line 298, we're creating the pyamqp/uamqp Message from the AmqpAnnotatedMessage. Inside `EventData._from_message`, `event_data._message` will be set to this Message.
  2. `raw_amqp_message: AmqpAnnotatedMessage` optional param. This is optional b/c we call `EventData._from_message` in two places. 
    * One is here during sending.  This call would still work if we didn't pass in raw_amqp_message here. However, since we already have the raw_amqp_message, we pass it in so that it doesn't get re-constructed inside for efficiency.
    * The other is when we receive. In receive, we only pass in the message=uamqp/pyamqp Message to this `EventData._from_message` method.

Please lmk if anything was unclear or if you have any concerns. :)